schemas/core/components/configurator.json

```diff
@@ -0,0 +1,70 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "description": "Configurator schema to customize a booking option",
+  "type": "object",
+  "description": "Wrapper schema, with each of its properties a config that the user can customize",
+  "patternProperties": {
+    "^[a-z\\-]+$": {
+      "$ref": "#/definitions/config"
+    }
+  },
+  "definitions": {
+    "config": {
+      "type": "object",
+      "description": "A customization to the booking option",
+      "properties": {
+        "type": {
+          "description": "Describe the possible combination of choices user can make",
+          "enum": ["oneOf", "someOf", "allOf", "oneOrNoneOf", "someOrNoneOf"]
+        },
+        "name": {
+          "type": "string",
+          "minLength": 0,
+          "maxLength": 255
+        },
+        "choices": {
+          "type": "array",
+          "uniqueItems": true,
+          "description": "Set of choices for one customization",
+          "items": {
+            "$ref": "#/definitions/choice"
+          },
+          "contains": {
+            "allOf": [
+              { "$ref": "#/definitions/choice" },
+              { "required": "default" }
+            ]
+          }
+        }
+      },
+      "required": ["type", "name", "choices"]
+    },
+    "choice": {
+      "type": "object",
+      "description": "A choice for one customization",
+      "properties": {
+        "id": {
+          "type": "string"
+        },
+        "name": {
+          "type": "string",
+          "minLength": 0,
+          "maxLength": 255
+        },
+        "default": {
+          "type": "boolean"
+        },
+        "cost": {
+          "$ref": "./cost.json"
+        }
+      },
+      "required": ["id", "name", "cost"]
+    },
+    "selections": {
```

Comment:
I guess `selection` (isn't it about one selection? so one list, not list of lists)

Reply:
Also maybe let's name it `customerSelection`, so semantics is obvious

Reply:
ok i'll change that 👍